src/main/java/pl/coderslab/projectwybankuj/web/StartController.java

```diff
@@ -0,0 +1,233 @@
+        } else if (chooseInsurance.equals("no") && chooseServiceCharge.equals("yes")) {
+            loans = loanRepository.findAllByParametersWithoutInsurance(amount, creditPeriod, age);
+        } else if (chooseServiceCharge.equals("no") && chooseInsurance.equals("yes")) {
+            loans = loanRepository.findAllByParametersWithoutServiceCharge(amount, creditPeriod, age);
+        } else {
+            loans = loanRepository.findAllByParameters(amount, creditPeriod, age);
+        }
+        Map<Loan, Double> loansWithPayments = loanService.calculateLoanPayment(loans, amount, creditPeriod);
+
+        model.addAttribute("loanSimulation", loansWithPayments);
+        model.addAttribute("amount", amount);
+        model.addAttribute("creditPeriod", creditPeriod);
+        model.addAttribute("chooseServiceCharge", chooseServiceCharge);
+        model.addAttribute("chooseInsurance", chooseInsurance);
+        model.addAttribute("age", age);
+
+        return "calculateloan";
+    }
+
+    @GetMapping("/loanDetails")
+    public String getLoanDetails(@RequestParam Long loanId, @RequestParam int amount, @RequestParam int creditPeriod,
+                                 @RequestParam String chooseServiceCharge, @RequestParam String chooseInsurance,
+                                 @RequestParam int age, Model model) {
+        Loan loan = loanRepository.findById(loanId).orElse(new Loan());
+        model.addAttribute("loan", loan);
+
+        double payment = loanService.calculateChoosenLoanPayment(loan, amount, creditPeriod);
+        model.addAttribute("payment", payment);
+
+        double serviceCharge = loanService.calculateServiceCharge(loan, amount);
+        model.addAttribute("serviceCharge", serviceCharge);
+
+        double insurance = loanService.calculateInsurance(loan, amount);
+        model.addAttribute("insurance", insurance);
+
+        double interests = loanService.calculateInterestsCost(loan, amount, creditPeriod, payment, serviceCharge, insurance);
+        model.addAttribute("interests", interests);
+
+        double totalCost = loanService.calculateTotalCost(loan, amount, creditPeriod, payment);
+        model.addAttribute("totalCost", totalCost);
+
+        model.addAttribute("amount", amount);
+        model.addAttribute("creditPeriod", creditPeriod);
+        model.addAttribute("serviceCharge", serviceCharge);
+        model.addAttribute("insurance", insurance);
+        model.addAttribute("age", age);
+        model.addAttribute("chooseServiceCharge", chooseServiceCharge);
+        model.addAttribute("chooseInsurance", chooseInsurance);
+
+        return "loandetails";
+    }
+
+    @GetMapping("/mortgageParameters")
+    public String getMortgageParameters() {
+        return "mortgageparameters";
+    }
+
+    @PostMapping("/mortgageParameters")
+    public String postMortgageParameters(@RequestParam int amount, @RequestParam int creditPeriod,
```

Comment:
Czy nie łatwiej i ładniej byłoby przekazać tutaj obiekt? Tak jak robiliśmy to na zajeciach i wyciągać z niego dane?